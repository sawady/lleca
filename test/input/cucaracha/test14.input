
fun f(h : Int, o : Int) {
  l := 108
  a := 97
  enter := 10
  putChar(h)
  putChar(o)
  putChar(l)
  putChar(a)
  putChar(enter)
}

fun main() {
  f(104, 111)
  f(72, 79)
}
