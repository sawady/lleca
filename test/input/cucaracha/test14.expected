Cons(
  Function(
    f,
    Cons(
      Parameter(
        h,
        Int
      ),
      Cons(
        Parameter(
          o,
          Int
        ),
        Nil
      )
    ),
    Unit,
    Cons(
      StmtAssign(
        l,
        ExprConstNum(
          108
        )
      ),
      Cons(
        StmtAssign(
          a,
          ExprConstNum(
            97
          )
        ),
        Cons(
          StmtAssign(
            enter,
            ExprConstNum(
              10
            )
          ),
          Cons(
            StmtCall(
              putChar,
              Cons(
                ExprVar(
                  h
                ),
                Nil
              )
            ),
            Cons(
              StmtCall(
                putChar,
                Cons(
                  ExprVar(
                    o
                  ),
                  Nil
                )
              ),
              Cons(
                StmtCall(
                  putChar,
                  Cons(
                    ExprVar(
                      l
                    ),
                    Nil
                  )
                ),
                Cons(
                  StmtCall(
                    putChar,
                    Cons(
                      ExprVar(
                        a
                      ),
                      Nil
                    )
                  ),
                  Cons(
                    StmtCall(
                      putChar,
                      Cons(
                        ExprVar(
                          enter
                        ),
                        Nil
                      )
                    ),
                    Nil
                  )
                )
              )
            )
          )
        )
      )
    )
  ),
  Cons(
    Function(
      main,
      Nil,
      Unit,
      Cons(
        StmtCall(
          f,
          Cons(
            ExprConstNum(
              104
            ),
            Cons(
              ExprConstNum(
                111
              ),
              Nil
            )
          )
        ),
        Cons(
          StmtCall(
            f,
            Cons(
              ExprConstNum(
                72
              ),
              Cons(
                ExprConstNum(
                  79
                ),
                Nil
              )
            )
          ),
          Nil
        )
      )
    ),
    Nil
  )
)
